components/Domain.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+    namespace Components;
+
+use InvalidArgumentException;
+
+/**
+     * Class used to retrieve different domain related strings, such as factory name
+     */
+    class Domain {
+
+        /**
+         * Name of the domain in plural
+         *
+         * @var string
+         */
+        private $domainPlural;
+
+        /**
+         * Name of the domain in singular
+         *
+         * @var string
+         */
+        private $domainSingular;
+
+        /**
+         * Name of the related factory
+         *
+         * @var string
+         */
+        private $factoryName;
+
+        /**
+         * Translated equivalent of the domain name
+         * 
+         * As manual translation is yet to be introduced, the translated version of the domain has to be stored manually
+         *
+         * @var string
+         */
+        private $translation;
+
+        /**
+         * A clause variant of the translated domain name
+         * 
+         * It is used in template pages(add, edit and delete)
+         *
+         * @var string
+         */
+        private $translationClause;
+
+        /**
+         * @param string $domainPlural
+         */
+        public function __construct(string $domainPlural) {
+            $this->domainPlural = $domainPlural;
+        }
+
+        /**
+         * Sets domain name in singular
+         *
+         * @param string $domainSingular
+         * @return self
+         */
+        public function setDomainSingular(string $domainSingular):self {
+            $this->domainSingular = $domainSingular;
+
+            return $this;
+        }
+
+        /**
+         * Sets related factory name
+         *
+         * @param string $factoryName
+         * @return self
+         */
+        public function setFactoryName(string $factoryName):self {
+            if ($factoryName === "") {
+                throw new InvalidArgumentException("Factory name must be a valid string");
+            }
+
+            $this->factoryName = $factoryName;
+
+            return $this;
+        }
+
+        /**
+         * Sets translated versions of domain name
+         *
+         * @param string $translation - translated versions of domain name
+         * @param string $translationClause - clause version of domain name
+         * @return self
+         */
```

Comment:
Both translation and translation clause are set at the same time because if the translation exists then the clause for it exists as well.